mailpoet/lib/AutomaticEmails/WooCommerce/Events/PurchasedInCategory.php

```diff
@@ -126,18 +127,17 @@ public function scheduleEmail($orderId) {
     }
 
     $schedulingCondition = function($automaticEmail) use ($orderedProductCategories, $subscriber) {
-      $meta = $automaticEmail->getMeta();
+      $matchedCategories = $this->getProductCategoryIdsMatchingNewsletterTrigger($automaticEmail, $orderedProductCategories);
+      if (empty($matchedCategories)) {
+        return false;
+      }
 
-      if (empty($meta['option'])) return false;
       if ($this->repository->wasScheduledForSubscriber($automaticEmail->id, $subscriber->id)) {
-        $sentAllProducts = $this->repository->alreadySentAllProducts($automaticEmail->id, $subscriber->id, 'orderedProductCategories', $orderedProductCategories);
+        $sentAllProducts = $this->repository->alreadySentAllProducts($automaticEmail->id, $subscriber->id, 'orderedProductCategories', $matchedCategories);
```

Comment:
The original code checked for `$orderedProductCategories` here, whereas the original code for `PurchasedProduct` searched just for the matched categories, so I updated this one to be consistent with the `PurchasedProduct` version. Would this alter behavior in an undesirable way?

Reply:
At first glimpse it doesn't seem to have a negative impact on the behavior, the change is a logical improvement to me. Is this case covered by tests?

Reply:
It wasn't covered in the tests, and I've now added one. I'm glad I did because I'm pretty sure this represents a current prod bug:

1. Set up an email with a trigger of category 15 with a 1 day delay
2. Make a purchase with category 15
3. Email is queued and we mark that the newsletter was queued for category 15
4. Make a purchase with category 15 and 16
5. We see that the newsletter was already scheduled 
6. `$sentAllProducts` is `false`, because we haven't marked category 16 as sent
7. A duplicate email is queued